app/services/service_offering/add_to_portfolio_item.rb

```diff
@@ -61,10 +61,10 @@ def create_icon(icon_id)
       end
 
       svc = Catalog::CreateIcon.new(
-        :content        => OpenStruct.new(:tempfile => file),
-        :source_ref     => service_offering_icon.source_ref,
-        :source_id      => service_offering_icon.source_id,
-        :portfolio_item => @item
+        :content           => OpenStruct.new(:tempfile => file),
+        :source_ref        => service_offering_icon.source_ref,
+        :source_id         => service_offering_icon.source_id,
+        :portfolio_item_id => @item.id
```

Comment:
I'm guessing this changed because of the parameters mixin change?

Reply:
It changed because of the fact that Icon doesn't have a `:portfolio_item` reference anymore, now that it's `iconable` the service class figures out where the icon belongs, here: https://github.com/lindgrenj6/catalog-api/blob/39fe32d7c1ae06761965c4eee4150ca72327291b/app/services/catalog/create_icon.rb#L9-L13

Since this is coming from topology it'll always be on a product, so we assume it's going to be on a PortfolioItem.